modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaGroup.es.js

```diff
@@ -0,0 +1,368 @@
+		const {criteria, editing, groupId, onMove, supportedConjunctions} = this.props;
+
+		return (
+			<Fragment>
+				<DropZone
+					groupId={groupId}
+					index={index}
+					onCriterionAdd={this._handleCriterionAdd}
+					onMove={onMove}
+				/>
+
+				{editing ?
+					<ClayButton
+						className="btn-sm conjunction-button"
+						label={this._getConjunctionLabel(
+							criteria.conjunctionName,
+							supportedConjunctions
+						)}
+						onClick={this._handleConjunctionClick}
+					/> :
+					<div className="conjunction-label">
+						{this._getConjunctionLabel(
+							criteria.conjunctionName,
+							supportedConjunctions
+						)}
+					</div>
+				}
+
+				<DropZone
+					before
+					groupId={groupId}
+					index={index}
+					onCriterionAdd={this._handleCriterionAdd}
+					onMove={onMove}
+				/>
+			</Fragment>
+		);
+	}
+
+	_renderCriterion = (criterion, index) => {
+		const {
+			editing,
+			groupId,
+			modelLabel,
+			onMove,
+			root,
+			supportedConjunctions,
+			supportedOperators,
+			supportedProperties,
+			supportedPropertyTypes
+		} = this.props;
+
+		const classes = getCN(
+			'criterion',
+			{
+				'criterion-group': criterion.items
+			}
+		);
+
+		return (
+			<div className={classes}>
+				{criterion.items ? (
+					<this.NestedCriteriaGroupWithDrag
+						criteria={criterion}
+						editing={editing}
+						groupId={criterion.groupId}
+						index={index}
+						modelLabel={modelLabel}
+						onChange={this._handleCriterionChange(index)}
+						onMove={onMove}
+						parentGroupId={groupId}
+						supportedConjunctions={supportedConjunctions}
+						supportedOperators={supportedOperators}
+						supportedProperties={supportedProperties}
+						supportedPropertyTypes={supportedPropertyTypes}
+					/>
+				) : (
+					<CriteriaRow
+						criterion={criterion}
+						editing={editing}
+						groupId={groupId}
+						index={index}
+						modelLabel={modelLabel}
+						onAdd={this._handleCriterionAdd}
+						onChange={this._handleCriterionChange(index)}
+						onDelete={this._handleCriterionDelete}
+						onMove={onMove}
+						root={root}
+						supportedConjunctions={supportedConjunctions}
+						supportedOperators={supportedOperators}
+						supportedProperties={supportedProperties}
+						supportedPropertyTypes={supportedPropertyTypes}
+					/>
+				)}
+
+				<DropZone
+					groupId={groupId}
+					index={index + 1}
+					onCriterionAdd={this._handleCriterionAdd}
+					onMove={onMove}
+				/>
+			</div>
+		);
+	}
+
+	render() {
+		const {
+			connectDragPreview,
+			connectDragSource,
+			criteria,
+			dragging,
+			editing,
+			groupId,
+			onMove,
+			root
+		} = this.props;
+
+		const classes = getCN(
+			'criteria-group-root',
+			{
+				'criteria-group-item': !root,
+				'criteria-group-item-root': root,
+				'dnd-drag': dragging
+			}
+		);
+
+		return connectDragPreview(
+			<div
+				className={classes}
+			>
+				{this._isCriteriaEmpty() ?
+					<EmptyDropZone
+						index={0}
```

Comment:
Are these indexes specifically for the drag/drop functionality? I wonder if it might be best to name it something like `dropIndex` just for a bit of clarity.

Reply:
That's a good idea. I'll rename. 👍